ubclaunchpad/ubc-degree-navigator
# Connect FE/BE for Uploading Course History

## Describe your changes
The pages related to course history are no longer hardcoded!! 

Users can choose to manually input all the courses they've taken or upload a screenshot of their course history from SSC. If the user chooses the upload option, the image is parsed and their courses will be displayed on the following `/manualCourseHistory` page. On the `/manualCourseHistory` page, one can add or delete courses as they please. Upon clicking the next button, these courses will be saved to the database.

#### Screenshot 
> This can be FE screenshot, DB screenshot, API call from network tab, etc.
![image](https://github.com/ubclaunchpad/ubc-degree-navigator/assets/115377855/bb9428f0-76f6-4924-95d0-827f049fbbf1)

Note: courses 13-16 in the screenshot above have `credit_counted = 0` because they are still in progress, and SSC displays the number of credits as 0 for those courses.

Note 2: as we don't have authentication set up, the `user_id` is hardcoded to be 1.

Note 3: when you manually enter a course, its `credit_counted` is hardcoded to be 3. However, the `credit_counted` is accurate for courses that were obtained from a transcript upload, as the number of credits are scraped from the transcript directly.

## How do I test
> Provide brief steps on how to test these changes. 
1. Run the react-app and go server
2. Upload a screenshot of your transcript found under the 'your grades summary' page on SSC 
3. Advance to the next page once the next button is enabled (the button is only enabled once the endpoint response is ready, im looking to add a loading icon to make that more clear)
4. Edit your course history by adding or deleting courses + adding courses in a new academic year (to make sure nothing breaks)
5. Click on the next button and open the `gorm.db` file to check that your course history + edits are saved 

## Checklist before requesting a review
- [ ] My base and comparing branches are correct
